### PR TITLE
Add Au test for TSVConnFdCreate().

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by ywkaras** [comment]
For some reason the read VIO doesn't get a READ_READY event until the connection is closed by the loopback thread:
```
[Sep  8 22:19:59.357] traffic_server DIAG: (TSVConnFd) initializing plugin
[Sep  8 22:19:59.357] traffic_server DIAG: (TSVConnFd) Plugin registration succeeded.
[Sep  8 22:19:59.358] traffic_server DIAG: (TSVConnFd) bind() with TCP port 61005
[Sep  8 22:19:59.399] [ET_NET 11] DIAG: (TSVConnFd) Send_to_vconn constructor: inst=0x7f2c180c8f10
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Send_to_vconn: status=0
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Send_to_vconn: status=0
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Send_to_vconn: status=2
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) ~Ramp_send()
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Recv_from_vconn: Event=100 inst=0x7f2c180d40e0
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Ramp_test: read avail=20000
[Sep  8 22:19:59.408] [ET_NET 11] DIAG: (TSVConnFd) Ramp_test: read avail=0
Traffic Server 10.0.0 Sep  8 2022 21:18:42 c1287bc2.home-wkaras.bf2.ows.oath.cloud
traffic_server: using root directory '/home/wkaras/REPOS/TS/tests/_sandbox/TSVConnFd/ts'
TSVConnFd: end Ramp_test::_loopback_data_thread_func()
[Sep  8 22:19:59.409] [ET_NET 11] DIAG: (TSVConnFd) Recv_from_vconn: Event=104 inst=0x7f2c180d40e0
[Sep  8 22:19:59.409] [ET_NET 11] DIAG: (TSVConnFd) ~Ramp_recv()
```

**Comment by ywkaras** [comment]
The review could be split between two reviewers.  One could review TSVConnFd.cc lines 1 - 429.  The other could review the rest of that file and the other files.

**Comment by ywkaras** [comment]
[approve ci autest]

File: tests/gold_tests/pluginTest/TSVConnFd/TSVConnFd.test.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    f'curl --verbose --ipv4 --header "Host:myhost.test" http://localhost:{ts.Variables.port}/'
+)
+tr.Processes.Default.ReturnCode = 0
+
+
+# Give tests up to 10 seconds to complete.
+#
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = (
+    "N=10 ; while ((N > 0 )) ; do " +
```
Comment (moonchen) [inline tests/gold_tests/pluginTest/TSVConnFd/TSVConnFd.test.py:75]: This script doesn't work on dash (default for /bin/sh on Ubuntu).  Suggested change:

```
"N=10 ; while [ $$N -gt 0 ] ; do " +
"if [ ! -f " + InProgressFilePathspec + " ] ; then exit 0 ; fi ; sleep 1 ; N=$$(( N-1 )) ; " +
"done ; echo 'TIMEOUT' ; exit 1"

**Review by moonchen — CHANGES_REQUESTED** [review]

File: tests/gold_tests/pluginTest/TSVConnFd/TSVConnFd.test.py (hunk 1 above)
Comment (ywkaras) [reply tests/gold_tests/pluginTest/TSVConnFd/TSVConnFd.test.py]: OK did it.

**Review by moonchen — APPROVED** [review]

**Pull request merged**